### ISSUE TITLE
Install failure of cupy==1.0.0.1 on Ubuntu 14.04, Cuda 8.0.61, Cudnn 6.0.21

### ISSUE THREAD

**Issue by wkentaro** [cupy/cupy#115]
```
$ echo $PATH
/usr/local/cuda/bin:/home/wkentaro/.local/bin:/usr/local/bin:/usr/local/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
$ echo $LD_LIBRARY_PATH
/usr/local/cuda/lib64:
$ echo $CFLAGS
-I/usr/local/cuda/include
$ echo $LDFLAGS
-L/usr/local/cuda/lib64
$ sudo pip install cupy --no-cache-dir -vvv
...
    building 'cupy.cuda.thrust' extension
    error: unknown file type '.cu' (from 'cupy/cuda/cupy_thrust.cu')
    Running setup.py install for cupy: finished with status 'error'
Cleaning up...
```

[full log](https://gist.github.com/wkentaro/cb01ad14e1702f9c34165e5c16a7ed9b)

It says below unexpectedly.

```
    **************************************************
    *** WARNING: nvcc not in path.
    *** WARNING: Please set path to nvcc.
    **************************************************
```

Do anyone have some idea to fix this? It worked before I upgrade the cuda to 8.0.61 from 8.0.5X.

**Comment by wkentaro** [comment]
I confirmed it works if I use the master branch, so maybe this problem is fixed recently (possibly https://github.com/cupy/cupy/pull/91 ?)

**Comment by unnonouno** [comment]
Did you use `sudo`?

**Comment by unnonouno** [comment]
From cupy 1.0.0, it install thrust and thrust is required to be build with `nvcc` command on setup time.

**Comment by wkentaro** [comment]
Yes, I use `sudo`. (Sorry, I updated the above comment).

**Comment by synetkim** [comment]
I had the same issue, and "install cupy for developer" resolves the issue.

$ pip install -U cython
$ cd /path/to/cupy/source
$ pip install -e .

**Comment by stephantul** [comment]
The `pip install -e .` command works on a macbook pro with OSX (no cuda), while installing from source (e.g. `python setup.py install`) didn't. Cheers!

**Comment by shirayu** [comment]
I faced the same problem on Ubuntu 16.04 where ``nvcc`` doesn't exist not in root's ``$PATH`` but in users' ``$PATH``.

According to ``man sudo``, ``sudo``  ignores users' ``$PATH``.
(It is the same when ``-E`` option is given to ``sudo``.)
Check that ``sudo env | grep ^PATH=`` is differ from ``env | grep ^PATH=``.

```
Note, however, that the actual PATH environment variable is not modified and is passed unchanged to the program that sudo executes
```


``sudo`` also does not inherit users' environment variables if ``env_reset`` is enabled in ``/etc/sudoers``.

With ``-E`` option, they are inherited except for ``$PATH``

So, the change of #91 to use ``$CUDA_HOME`` seems to be good.
Here again, ``-E`` option for ``sudo`` is need in some cases.

**Comment by knorth55** [comment]
Now I can successfully install `cupy-1.0.1` with `sudo pip install cupy`.

**Comment by wkentaro** [comment]
Yeah, now it is fixed by release of 1.0.1. Thanks.
